util/integermath.go

```diff
@@ -0,0 +1,65 @@
+package util
+
+// Max returns the minimum value of inputs x, y
+func Max(x int, y int) int {
+	if x > y {
+		return x
+	}
+	return y
+}
+
+// Min returns the minimum value of inputs x, y
+func Min(x int, y int) int {
+	if x < y {
+		return x
+	}
+	return y
+}
+
+// Ceil computes the ceiling of x/y for x, y being integers
+func Ceil(x int, y int) int {
+	if x == 0 {
+		return 0
+	}
+	return 1 + ((Abs(x) - 1) / Abs(y))
+}
+
+func Abs(x int) int {
+	if x < 0 {
+		return -x
+	}
+	return x
+}
+
+// Log2Ceil computes the integer logarithm with ceiling for up to 64 bit ints
+// Translated from https://www.appsloveworld.com/c/100/6/compute-fast-log-base-2-ceiling
+func Log2Ceil(value int) int {
```

Comment:
In theory, this can be replaced with code that uses `bits.OnesCount` and `bits.LeadingZeros`.
We also should probably accept uint64 when doing math like that for consistent crossplatform results.

Reply:
Handled in commit [882dedf](https://github.com/filecoin-project/go-data-segment/pull/2/commits/882dedffc046b7f5205bbb4e60686caa0c52cdf8)